src/main/java/hudson/plugins/ec2/SlaveTemplate.java

```diff
@@ -857,12 +904,13 @@ private void logProvisionInfo(String message) {
         LOGGER.info(this + ". " + message);
     }
 
-    HashMap<RunInstancesRequest, List<Filter>> makeRunInstancesRequestAndFilters(int number, AmazonEC2 ec2) throws IOException {
-        return makeRunInstancesRequestAndFilters(number, ec2, true);
+    HashMap<RunInstancesRequest, List<Filter>> makeRunInstancesRequestAndFilters(Image image, int number, AmazonEC2 ec2) throws IOException {
+        return makeRunInstancesRequestAndFilters(image, number, ec2, true);
```

Comment:
I don't think this concrete change should be an issue since I don't think this package is used out of this plugin, but it's a good practice in the community to deprecate the method and create the new method with the signature changes. The deprecated method can be removed in some versions.
This good practice is applied to all non-private and non-restricted methods just in case there is any proprietary plugin from any user that is extending this code. -> FYI @froblesmartin

Reply:
Good call. I hadn't thought of any of the methods besides the constructor being used outside of the package.